### PR TITLE
keep comment appearing between parameter's name and its type

### LINKED ISSUE #2976: rustfmt removes a block comment in an argument position
[Input](https://play.rust-lang.org/?gist=a94a6a5c67ee2266058f0378d167d77f&version=nightly&mode=debug&edition=2015):

```rust
fn a(_ /*comment*/: u8) {}
fn b(/*comment*/ _: u8) {}

fn main() {}

```

Expected output (no changes):

```rust
fn a(_ /*comment*/: u8) {}
fn b(/*comment*/ _: u8) {}

fn main() {}

```

Real output (on of the comments is removed):

```rust
fn a(_: u8) {}
fn b(/*comment*/ _: u8) {}

fn main() {}
```

Replacing `_` with normal identifier doesn't help.

I guess it's related to https://github.com/rust-lang-nursery/rustfmt/issues/1096.

### PR DESCRIPTION
**Pull request by scampi**
Close #2976

### CODE REVIEW

File: src/items.rs [hunk 1]
```diff
@@ -1828,6 +1825,60 @@ fn is_empty_infer(ty: &ast::Ty, pat_span: Span) -> bool {
...
+        }
+    };
+    let trailing_space = |com: String| {
+        if com.is_empty() {
+            String::new()
+        } else {
+            format!("{} ", com)
+        }
+    };
+    match (comment_before_colon, comment_after_colon) {
```
Comment (topecongiro) [inline src/items.rs]: It's more or less a matter of a taste...however I would rather write these as something like the following:

```rust
comment_before_colon.map(leading_space).unwrap_or(String::new)
```

You may need to change the signature of closures to `|com: &str|`.

Comment (scampi) [reply src/items.rs]: Thanks for the suggestion, I like it! I tweaked it a bit but I kept the spirit.

Comment (topecongiro) [reply src/items.rs]: Nice, even better :)

**Comment by topecongiro** [comment]
Thank you!

**Pull request merged**